lib/factorial.rb

```diff
@@ -1,4 +1,22 @@
 # Computes factorial of the input number and returns it
 def factorial(number)
-  raise NotImplementedError
+  # edge cases test - number is nil
+  if number == nil
+    raise ArgumentError
+  end
+
+  # edge case test, number is 0
+  if number == 0
+    return 1
+  end
+
+  # factorial: ! means factorial. n! = n * (n-1) if n > 0
+  # example: 5! = 5 * 4 * 3 * 2 * 1
+  n = number
+  while n > 1
```

Comment:
This could result in an infinite loop. Are the mini-tests provided passing for you?